tests/test_tracker.py

```diff
@@ -353,6 +353,59 @@ def dist(new_obj, tracked_obj):
     assert tracked_objects[0].id != obj_id
 
 
+def test_reid_hit_counter_reset():
+    #
+    # test that reid hit counter resets to None if it had started counting down but
+    # then the track was hit with an incoming detection
+    #
+
+    # simple reid distance
+    def dist(new_obj, tracked_obj):
+        return np.linalg.norm(new_obj.estimate - tracked_obj.estimate)
+
+    hit_counter_max = 2
+    reid_hit_counter_max = 2
+
+    tracker = Tracker(
+        distance_function="euclidean",
+        distance_threshold=1,
+        hit_counter_max=hit_counter_max,
+        initialization_delay=1,
+        reid_distance_function=dist,
+        reid_distance_threshold=5,
+        reid_hit_counter_max=reid_hit_counter_max,
+    )
+
+    # check that hit counters initialize correctly
+    tracked_objects = tracker.update([Detection(points=np.array([[1, 1]]))])
+    tracked_objects = tracker.update([Detection(points=np.array([[1, 1]]))])
+    assert len(tracked_objects) == 1
+    assert tracked_objects[0].hit_counter == 2
+    assert tracked_objects[0].reid_hit_counter == None
+
+    # check that object is still alive when hit_counter goes to 0
+    obj_id = tracked_objects[0].id
+    for _ in range(hit_counter_max):
+        tracked_objects = tracker.update()
+    assert len(tracked_objects) == 1
+    assert tracked_objects[0].hit_counter == 0
+    assert tracked_objects[0].reid_hit_counter is None
+
+    # check that object is alive and reid_hit_counter is None after being matched again
+    tracked_objects = tracker.update([Detection(points=np.array([[1, 1]]))])
+    assert len(tracked_objects) == 1
+    assert tracked_objects[0].hit_counter == 1
+    assert tracked_objects[0].reid_hit_counter is None
+
+    # check that after reid_hit_counter_max more updates, object still exists
+    for _ in range(reid_hit_counter_max + 2):
```

Comment:
+2 because with the bug,
- After `reid_hit_counter_max` updates, `reid_hit_counter` will be 0
- After `reid_hit_counter_max` + 1 updates, `reid_hit_counter` will be -1 but because of the bug, `hit_counter` will be positive at the end of the update call, so the tracked object will still be returned and alive
- At the start of the `reid_hit_counter_max` + 2 update, `reid_hit_counter` will be -1 and the tracked object will be marked as dead and won't be sent for normal matching